src/lib.rs

```diff
@@ -554,6 +563,21 @@ impl Contract {
             .set(json!({ get_devhub_community_account(&handle): data }));
     }
 
+    pub fn create_discussion(&mut self, handle: CommunityHandle, data: Value) {
+        require!(env::prepaid_gas() >= CREATE_DISCUSSION_GAS, "Require at least 60 Tgas");
+        // Post the discussion on the user social account
+        let promise = social_db_contract()
+            .with_unused_gas_weight(1)
+            .set(json!({ format!("{}", env::predecessor_account_id()): data }));
```

Comment:
That is not going to work. devhub contract does not have permission to post on behalf of the user. DevHub UI (BOS component) should post on behalf of the user and then call devhub to reshare it to the community discussions account.

Reply:
I've implemented the front end to call `grant_write_permission` first, the same way I did it in the test case.

Reply:
@Tguntenaar As a user, I don't want to grant DevHub to post anything on my Social Feed at any moment. Don't do that.

Reply:
> @Tguntenaar As a user, I don't want to grant DevHub contract to post anything on my behalf to Social Feed at any moment. Don't do that.

This is reasonable to the user, but the repost approach will not work without this assumption:

- Current UX ensures user only needs to submit his post once with one click to dicussions
- He needs to submit to social db contract for his feed, then repost to community discussion feed (a cross contract call to social db)
- But we cannot get social db cross contract call social db contract
- Nor can we do the post and repost as two transactions happen once on the gateway, because repost need the post's block height
- So the only viable option is Thomas implemented get devhub.near to cross contract call social db contract twice and write on user's behalf

Reply:
I don't see a strong requirement in making that in one function call from the user. I even suggest that we consider if we should have a way to re-share posts to other communities and even to the announcements feeds. Thus, I believe it is fine to post discussion post to the community by separating it into two transactions: (1) post to the user feed, (2) re-share to the discussion feed (if we will see that users have a broken flow here, we may just do it with our own Indexer of social.near catching new posts and calling devhub.near to reshare them to the discussions)

Reply:
@Tguntenaar Just to further clarify it with some pseudocode, I would expect to do the following on the BOS (frontend) side:

```js
const publishDiscussion = (text) => {
  Social.set(`${context.accountId}/post/main`, text, { onCommit: (data) => {
    Near.call("devhub.near", "create_discussion", { community_handle: "...", near_social_post_block_height: blockHeight });
  }})
}
```

The challenge here is to get the `blockHeight` of the published post. Maybe you will have to do 'Social.get(`${context.accountId}/post/main`, { options: { with_block_height: true } })' (see options [here](https://github.com/NearSocial/social-db/blob/b4a023d234d0f24a2cf0d4b7998ea485f90f576e/contract/src/api.rs#L18))